core/domain/user_services.py

```diff
@@ -931,24 +1010,225 @@ def create_new_user(gae_id, email):
+            attribute.
+    """
+
+    def _create_new_profile_transactional(
+            user_settings, user_auth_details):
+        """Save user models for new users as a transaction.
+
+        Args:
+            user_settings: UserSettings. The user settings domain object
+                corresponding to the newly created user.
+            user_auth_details: UserAuthDetails. The user auth details domain
+                object corresponding to the newly created list of users.
+        """
+        _save_user_auth_details(user_auth_details)
+        _save_user_settings(user_settings)
+
+    # As new profile user creation is done by a full (parent) user only.
+    parent_user_auth_details = get_auth_details_by_gae_id(gae_id, strict=True)
+    if parent_user_auth_details.pin is None:
+        raise Exception(
+            'Pin must be set for a full user before creating a profile.')
+    parent_user_id = parent_user_auth_details.user_id
+    user_settings_list = []
+    for modifiable_user_data in modifiable_user_data_list:
+        if modifiable_user_data.user_id:
+            raise Exception('User id cannot already exist for a new user.')
+        user_id = user_models.UserSettingsModel.get_new_id('')
+        user_settings = UserSettings(
+            user_id, gae_id, email, feconf.ROLE_ID_LEARNER,
+            preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE]
+        )
+        user_settings.populate_from_user_data(modifiable_user_data)
+
+        user_auth_details = UserAuthDetails(
+            user_id, None, modifiable_user_data.pin, parent_user_id)
+
+        transaction_services.run_in_transaction(
+            _create_new_profile_transactional, user_settings,
+            user_auth_details
+        )
+        user_settings_list.append(user_settings)
+    return user_settings_list
+
+
+def update_multiple_users_data(modifiable_user_data_list):
+    """Updates user settings and user auth model details for the users
+    specified in the modifiable_user_data_list.
+
+    Args: user_details_change_list: list(ModifiableUserData). The list of
+            modifiable_user_data entries corresponding to the users whose
+            data has to be updated.
+
+    Raises:
+        ValidationError. None or empty value is provided for display
+            alias attribute.
+        Exception. None or empty value is provided for user_id.
+    """
+    user_settings_list = []
+    user_auth_details_list = []
+    for modifiable_user_data in modifiable_user_data_list:
+        user_id = modifiable_user_data.user_id
+        if not user_id:
+            raise Exception(
+                'A valid user id must exist for all the users to be updated.')
+        user_settings = get_user_settings(user_id, strict=True)
+        user_settings.populate_from_user_data(modifiable_user_data)
+        user_settings_list.append(user_settings)
+
+        user_auth_details = get_auth_details_by_user_id(user_id, strict=True)
```

Comment:
Sorry for the get_multi() part. I missed it, but I will fix it carefully at all places in my PR.

Regarding PIN:
**(Approach 1 - current one)**
I did not re-think a lot (or got it double checked by @vinitamurthi) about this while changing the code, but my reasoning for keeping the PIN:
(a) when we shift to another device, we do want to have the same PIN there also for all the profiles right?
(b) Even if we don't want that, but still we would want **full user**'s PIN to be the **same** on the new device - because acc to the current concept administrator pin has to be set before any profile can be created.

**(Approach 2 - another possibility - seems reasonable enough to me)**
(c) If (a) is not necessary and we decide not to sync the PIN then we can inform the user that the PIN is local only. Upon switching to a new device, first the **user will have to set an admin (full user's) PIN before he can start using the account if that account already has profiles** - this will be **ensured from the android app** only.
(d) And the pin for None of the fetched profiles users (if any) would be set. It would have to be reset here.

Due to change in the way we are syncing profiles now, I am not sure what the user expectation will be. Hence can't do the pros/cons analysis myself. Looking for comments on this. Thanks!

Reply:
Yup, I get that we want to store it, and that we should store it somewhere. But why are we doing so in the UserAuth model (rather than UserSettings) if we're not using it to log on in the server, but rather just treating the server as a storage space for it?

Reply:
Okay, we will store the PIN on the server. 
But I just wanted to double-check on your second statement on PIN - " if we're not using it to log on in the server, but rather just treating the server as a storage space for it". Reason being since we updated our thinking on syncing of android profiles, I am a bit confused if the auth validation is going to be done or not for profiles as @BenHenning mentioned in this [comment](https://github.com/oppia/oppia/pull/10105#discussion_r463375498) **using PIN** in the backend.
@BenHenning I think you mentioned once that a session based authentication (or something like that) will be used for actually implementing auth for profiles (and for all android request basically), so can I move PIN back to the UserSettingsModel now if PIN is just being treated as server storage space. Just double checking. Thanks!

Reply:
To answer @seanlip's question, I believe we decided to store it in UserAuth because at the end of the day it is an authentication detail of the user, even if we won't use it to authenticate the user. Its not exactly a user setting but its authentication information

Reply:
Yup, but that was the with the old framing and I'm not sure that that makes sense with the current one. Let me talk with @BenHenning about it later today and one of us can get back to this thread with an update -- thanks!

Reply:
Okay I will move `PIN` to the `UserSettingsModel` as it isn't going be used for authentication at the backend side (@BenHenning confirmed it today on the email thread).